include/z64pause.h

```diff
@@ -26,6 +26,35 @@ typedef enum {
 #define PAUSE_EQUIP_BUFFER_SIZE sizeof(u16[PAUSE_EQUIP_PLAYER_HEIGHT][PAUSE_EQUIP_PLAYER_WIDTH])
 #define PAUSE_PLAYER_SEGMENT_GAMEPLAY_KEEP_BUFFER_SIZE 0x5000
 
+typedef enum {
+    /*  0 */ PAUSE_STATE_OFF,
+    /*  1 */ PAUSE_STATE_WAIT_LETTERBOX, // Request no letterboxing and wait for it.
+    /*  2 */ PAUSE_STATE_WAIT_BG_PRERENDER, // Wait for the pause background prerender to be done.
+    /*  3 */ PAUSE_STATE_INIT, // Load data and initialize/setup various things.
+    /*  4 */ PAUSE_STATE_OPENING_1, // Animate the pause menu coming together with rotations and other animations.
+    /*  5 */ PAUSE_STATE_OPENING_2, // Finish some animations for opening the menu.
+    /*  6 */ PAUSE_STATE_MAIN, // Pause menu ready for player inputs.
+    /*  7 */ PAUSE_STATE_SAVE_PROMPT,  // Save prompt in the pause menu
+    /*  8 */ PAUSE_STATE_8,
+    /*  9 */ PAUSE_STATE_9,
+    /* 10 */ PAUSE_STATE_10,
+    /* 11 */ PAUSE_STATE_11,
+    /* 12 */ PAUSE_STATE_12,
+    /* 13 */ PAUSE_STATE_13,
+    /* 14 */ PAUSE_STATE_14,
+    /* 15 */ PAUSE_STATE_15,
+    /* 16 */ PAUSE_STATE_16,
+    /* 17 */ PAUSE_STATE_17,
+    /* 18 */ PAUSE_STATE_CLOSING, // Animate the pause menu closing
+    /* 19 */ PAUSE_STATE_UNPAUSE
```

Comment:
whats the difference here? is "unapuse" that state the whole time while the game is unpaused and the player is playing the game? in that cause "unpaused" might be better, or "inactive" if main is changed to "active"

Reply:
Unpause is the state that deals with unpausing, when the game isn't paused the state is OFF

I didn't realize state_unpause could be understood as state_unpaused, would state_unpausing be better?

Reply:
it would, but at a cursory glance (without diving into to the code) am wondering how that differs from closing. the menu closing is usually associated with unpausing in general, at a very high level

Reply:
CLOSING is the animation of the pause menu going (rotating) away,
UNPAUSE is do things to resume normal play state execution like restoring the object buffer contents, segtovirtual the collision headers, and some various other stuff

Reply:
I see
maybe a name centered around "resume" or "restore" would be nice, just to try to be more specific about what its doing. havent put much thought into it though, will let you know if I come up with anything

Reply:
`UNPAUSE` -> `RESUME_GAMEPLAY` and added a comment in the enum